o2a/mappers/fs_mapper.py

```diff
@@ -112,47 +116,79 @@ class FsMapper(ActionMapper):
     Converts a FS Oozie node to an Airflow task.
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        dag_name: str,
+        property_set: PropertySet,
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        **kwargs,
+    ):
+        super().__init__(
+            oozie_node=oozie_node,
+            name=name,
+            property_set=property_set,
+            trigger_rule=trigger_rule,
+            dag_name=dag_name,
+            **kwargs,
+        )
+        self.oozie_node = oozie_node
+        self.trigger_rule = trigger_rule
         self.tasks: List[Task] = []
 
     def on_parse_node(self):
         super().on_parse_node()
         self.tasks = self.parse_tasks()
 
-    def parse_tasks(self):
+    def parse_tasks(self) -> List[Task]:
         if not list(self.oozie_node):
             return [Task(task_id=self.name, template_name="dummy.tpl", trigger_rule=self.trigger_rule)]
 
-        return [self.parse_fs_action(i, node) for i, node in enumerate(self.oozie_node)]
-
-    def to_tasks_and_relations(self):
+        tasks: List[Task] = []
+        index = 0
+        for node in self.oozie_node:
+            task = self.parse_fs_action(index, node)
```

Comment:
Maybe it is better to prevent such a case by moving the condition check earlier?

Reply:
I have a ready patch that improves it.

Reply:
My patch is available: 
https://github.com/GoogleCloudPlatform/oozie-to-airflow/commit/d2691556b3c8b5840b626c813b6c474277c17ba1

Reply:
Maybe instead of such patches we should add a commit to the PR itself? I think it would be clearer and easier to review.

Reply:
I prefer not to change this PR without the approval of Jarek.

Reply:
Let's discuss one-by-one and reach consensus what we do. I plan to squash and possibly re-split the change to multiple commits to separate out some changes so we can do whatever we decide as appropriate.